flake.nix

```diff
@@ -0,0 +1,57 @@
+{
```

Comment:
I know nix is very popular to create reproducible environments. I've only had nasty experiences with it though. I don't oppose having it but as Jose suggests, this needs to be documented.